Samples/dot-net-demo/dot-net-demo/Controllers/HomeController.cs

```diff
@@ -62,61 +62,68 @@ public IActionResult LRLogin([FromBody] EmailLoginModel emailLoginModel)
-                    var breachdetecteddateString = apiresponse.Response.Profile.CustomFields[BreachVeriable];
+                    if (apiresponse.Response != null && apiresponse.Response.Profile != null &&
+                        apiresponse.Response.Profile.CustomFields != null
+                        && apiresponse.Response.Profile.CustomFields.ContainsKey(BreachVeriable)
+                        && !string.IsNullOrEmpty(apiresponse.Response.Profile.CustomFields[BreachVeriable])
+                        && apiresponse.Response.Profile.LastPasswordChangeDate.HasValue)
+                    {
+                        var breachdetecteddateString = apiresponse.Response.Profile.CustomFields[BreachVeriable];
 
-                    DateTime.TryParse(breachdetecteddateString, out DateTime breachDate);
+                        DateTime.TryParse(breachdetecteddateString, out DateTime breachDate);
 
-                    if (apiresponse.Response.Profile.LastPasswordChangeDate > breachDate)
-                    {
-                        passwordChangedAfterBreach = true;
+                        if (apiresponse.Response.Profile.LastPasswordChangeDate > breachDate)
+                        {
+                            passwordChangedAfterBreach = true;
+                        }
                     }
-                }
 
-                if (!passwordChangedAfterBreach)
-                {
-                    try
+                    if (!passwordChangedAfterBreach)
                     {
-                        var payload = new { email = log.Email== "lruser123@mail7.io"?"test@gmail.com":log.Email };
-                        var _request = new RestRequest(Method.POST);
-                        _request.AddHeader("Accept", "*/*");
-                        _request.AddHeader("Content-Type", "application/json");
-                        _request.AddJsonBody(payload);
-                        _request.JsonSerializer = new RestSharp.Serialization.Json.JsonSerializer();
-                        var RestClient = new RestClient("https://openfaas-dev.div4.dev/function/jarvis-dark-web");
-                        RestClient.Authenticator = new HttpBasicAuthenticator("admin", "Yel9HPJAVZFT");
-                        var resp = RestClient.Execute(_request);
-
-                        if (resp.StatusCode == HttpStatusCode.OK && !string.IsNullOrWhiteSpace(resp.Content))
+                        try
                         {
-                            var updatemodel = new UserProfileUpdateModel { CustomFields = new Dictionary<string, string>() };
-                            updatemodel.CustomFields.Add(BreachVeriable, DateTime.UtcNow.ToString());
-                            var setCheckBreach = new AuthenticationApi().
-                                UpdateProfileByAccessToken(apiresponse.Response.Access_Token.ToString(), updatemodel);
-
-                            var currentDomain = $"{this.Request.Scheme}://{this.Request.Host}{this.Request.PathBase}";
```

Comment:
@hemant404 Are we sending the reset password link in both case?

Reply:
@anilswm Yes, it is being sent in both cases.